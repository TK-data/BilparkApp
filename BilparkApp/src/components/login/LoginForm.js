import React, { Component } from 'react';
import t from 'tcomb-form-native';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import { StyleSheet, View } from 'react-native';
import { Button, Text, Spinner } from 'native-base';
import { postUser, postCurrent, loginMail } from '../../actions/auth';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.props.postCurrent();
  }

  handleSubmit() {
    const value = this.form.getValue();
    if (value) {
      this.props.loginMail({ Epost: value.Epost });
      this.props.postUser(value.Epost, value.Passord);
    }
  }

  render() {

    const mailCheck = t.refinement(t.String, (email) => {
      const reg = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
      return reg.test(email);
    });

    const passwordCheck = t.refinement(t.String, (pass) => {
      return (pass.length >= 8);
    });

    const Login = t.struct({
      Epost: mailCheck,
      Passord: passwordCheck,
    });

    const Form = t.form.Form;

    if (this.props.isLoading) {
      return <View style={styles.container}><Spinner color="white" /></View>;
    }

    return (
      <View style={styles.container}>
        <Form
          ref={c => this.form = c}
          type={Login}
          options={this.props.formOptions}
          value={this.props.mail}
        />
        <Button
          bordered
          light
          onPress={() => {
            this.handleSubmit();
          }}
          style={styles.button}
        >
          <Text> Logg inn </Text>
        </Button>
        <Button
          bordered
          light
          onPress={() => this.props.registerScreen()}
          style={styles.button}
        >
          <Text>Registrer deg her</Text>
        </Button>
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isLoading: state.auth.isLoading,
    hasErrored: state.auth.hasErrored,
    user: state.auth.user,
    mail: state.loginMail,
    formOptions: state.loginOptions,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    postUser: (username, password) => dispatch(postUser(username, password)),
    postCurrent: () => dispatch(postCurrent()),
    loginMail: mail => dispatch(loginMail(mail)),
    registerScreen: () => {
      dispatch(NavigationActions.navigate({ routeName: 'Register' }));
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoginForm);


const styles = StyleSheet.create({
  container: {
  },
  input: {
    height: 37,
    width: 250,
  },
  text: {
    color: '#fff',
  },
  button: {
    alignSelf: 'center',
    marginBottom: 12,
  },
});
