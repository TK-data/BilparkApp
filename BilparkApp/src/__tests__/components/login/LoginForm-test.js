import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LoginForm from '../../../components/login/LoginForm';

const initialState = {
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: 'user',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing LoginForm', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <LoginForm />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
