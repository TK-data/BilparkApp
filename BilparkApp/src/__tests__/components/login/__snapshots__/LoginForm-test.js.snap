// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginForm Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    formOptions={undefined}
    hasErrored={false}
    isLoading={false}
    loginMail={[Function]}
    mail={undefined}
    postCurrent={[Function]}
    postUser={[Function]}
    registerScreen={[Function]}
    user="user"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Form
          i18n={
            Object {
              "add": "Add",
              "down": "↓",
              "optional": " (optional)",
              "remove": "✘",
              "required": "",
              "up": "↑",
            }
          }
          options={undefined}
          stylesheet={
            Object {
              "button": Object {
                "alignSelf": "stretch",
                "backgroundColor": "#48BBEC",
                "borderColor": "#48BBEC",
                "borderRadius": 8,
                "borderWidth": 1,
                "height": 36,
                "justifyContent": "center",
                "marginBottom": 10,
              },
              "buttonText": Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 18,
              },
              "checkbox": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "controlLabel": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
              },
              "dateTouchable": Object {
                "error": Object {},
                "normal": Object {},
              },
              "dateValue": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
              },
              "datepicker": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "errorBlock": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "fieldset": Object {},
              "formGroup": Object {
                "error": Object {
                  "marginBottom": 10,
                },
                "normal": Object {
                  "marginBottom": 10,
                },
              },
              "helpBlock": Object {
                "error": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
                "normal": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
              },
              "pickerContainer": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "open": Object {},
              },
              "pickerTouchable": Object {
                "active": Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#cccccc",
                },
                "error": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
                "normal": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
              },
              "pickerValue": Object {
                "error": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
                "normal": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
              },
              "select": Object {
                "error": Object {},
                "normal": Object {},
              },
              "textbox": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "notEditable": Object {
                  "backgroundColor": "#eeeeee",
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#777777",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
              },
              "textboxView": Object {
                "error": Object {},
                "normal": Object {},
                "notEditable": Object {},
              },
            }
          }
          templates={
            Object {
              "checkbox": [Function],
              "datepicker": [Function],
              "list": [Function],
              "select": [Function],
              "struct": [Function],
              "textbox": [Function],
            }
          }
          type={[Function]}
          value={undefined}
        />,
        <Styled(Button)
          bordered={true}
          light={true}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "marginBottom": 12,
            }
          }
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
             Logg inn 
          </Styled(Text)>
        </Styled(Button)>,
        <Styled(Button)
          bordered={true}
          light={true}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "marginBottom": 12,
            }
          }
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
            Registrer deg her
          </Styled(Text)>
        </Styled(Button)>,
      ],
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "i18n": Object {
            "add": "Add",
            "down": "↓",
            "optional": " (optional)",
            "remove": "✘",
            "required": "",
            "up": "↑",
          },
          "options": undefined,
          "stylesheet": Object {
            "button": Object {
              "alignSelf": "stretch",
              "backgroundColor": "#48BBEC",
              "borderColor": "#48BBEC",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 36,
              "justifyContent": "center",
              "marginBottom": 10,
            },
            "buttonText": Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 18,
            },
            "checkbox": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "controlLabel": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
            },
            "dateTouchable": Object {
              "error": Object {},
              "normal": Object {},
            },
            "dateValue": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
            },
            "datepicker": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "errorBlock": Object {
              "color": "#a94442",
              "fontSize": 17,
              "marginBottom": 2,
            },
            "fieldset": Object {},
            "formGroup": Object {
              "error": Object {
                "marginBottom": 10,
              },
              "normal": Object {
                "marginBottom": 10,
              },
            },
            "helpBlock": Object {
              "error": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "normal": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
            },
            "pickerContainer": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "open": Object {},
            },
            "pickerTouchable": Object {
              "active": Object {
                "borderBottomWidth": 1,
                "borderColor": "#cccccc",
              },
              "error": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
              "normal": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
            },
            "pickerValue": Object {
              "error": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
              "normal": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
            },
            "select": Object {
              "error": Object {},
              "normal": Object {},
            },
            "textbox": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "notEditable": Object {
                "backgroundColor": "#eeeeee",
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#777777",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
            },
            "textboxView": Object {
              "error": Object {},
              "normal": Object {},
              "notEditable": Object {},
            },
          },
          "templates": Object {
            "checkbox": [Function],
            "datepicker": [Function],
            "list": [Function],
            "select": [Function],
            "struct": [Function],
            "textbox": [Function],
          },
          "type": [Function],
          "value": undefined,
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": true,
          "children": <Styled(Text)
            virtual={undefined}
          >
             Logg inn 
          </Styled(Text)>,
          "light": true,
          "onPress": [Function],
          "style": Object {
            "alignSelf": "center",
            "marginBottom": 12,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Logg inn ",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": " Logg inn ",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": true,
          "children": <Styled(Text)
            virtual={undefined}
          >
            Registrer deg her
          </Styled(Text)>,
          "light": true,
          "onPress": [Function],
          "style": Object {
            "alignSelf": "center",
            "marginBottom": 12,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Registrer deg her",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Registrer deg her",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Form
            i18n={
              Object {
                "add": "Add",
                "down": "↓",
                "optional": " (optional)",
                "remove": "✘",
                "required": "",
                "up": "↑",
              }
            }
            options={undefined}
            stylesheet={
              Object {
                "button": Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#48BBEC",
                  "borderColor": "#48BBEC",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "height": 36,
                  "justifyContent": "center",
                  "marginBottom": 10,
                },
                "buttonText": Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontSize": 18,
                },
                "checkbox": Object {
                  "error": Object {
                    "marginBottom": 4,
                  },
                  "normal": Object {
                    "marginBottom": 4,
                  },
                },
                "controlLabel": Object {
                  "error": Object {
                    "color": "#a94442",
                    "fontSize": 17,
                    "fontWeight": "500",
                    "marginBottom": 7,
                  },
                  "normal": Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "fontWeight": "500",
                    "marginBottom": 7,
                  },
                },
                "dateTouchable": Object {
                  "error": Object {},
                  "normal": Object {},
                },
                "dateValue": Object {
                  "error": Object {
                    "color": "#a94442",
                    "fontSize": 17,
                    "marginBottom": 5,
                    "padding": 7,
                  },
                  "normal": Object {
                    "color": "#000000",
                    "fontSize": 17,
                    "marginBottom": 5,
                    "padding": 7,
                  },
                },
                "datepicker": Object {
                  "error": Object {
                    "marginBottom": 4,
                  },
                  "normal": Object {
                    "marginBottom": 4,
                  },
                },
                "errorBlock": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
                "fieldset": Object {},
                "formGroup": Object {
                  "error": Object {
                    "marginBottom": 10,
                  },
                  "normal": Object {
                    "marginBottom": 10,
                  },
                },
                "helpBlock": Object {
                  "error": Object {
                    "color": "#999999",
                    "fontSize": 17,
                    "marginBottom": 2,
                  },
                  "normal": Object {
                    "color": "#999999",
                    "fontSize": 17,
                    "marginBottom": 2,
                  },
                },
                "pickerContainer": Object {
                  "error": Object {
                    "borderColor": "#a94442",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "marginBottom": 4,
                  },
                  "normal": Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "marginBottom": 4,
                  },
                  "open": Object {},
                },
                "pickerTouchable": Object {
                  "active": Object {
                    "borderBottomWidth": 1,
                    "borderColor": "#cccccc",
                  },
                  "error": Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 44,
                  },
                  "normal": Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 44,
                  },
                },
                "pickerValue": Object {
                  "error": Object {
                    "fontSize": 17,
                    "paddingLeft": 7,
                  },
                  "normal": Object {
                    "fontSize": 17,
                    "paddingLeft": 7,
                  },
                },
                "select": Object {
                  "error": Object {},
                  "normal": Object {},
                },
                "textbox": Object {
                  "error": Object {
                    "borderColor": "#a94442",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  },
                  "normal": Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  },
                  "notEditable": Object {
                    "backgroundColor": "#eeeeee",
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#777777",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  },
                },
                "textboxView": Object {
                  "error": Object {},
                  "normal": Object {},
                  "notEditable": Object {},
                },
              }
            }
            templates={
              Object {
                "checkbox": [Function],
                "datepicker": [Function],
                "list": [Function],
                "select": [Function],
                "struct": [Function],
                "textbox": [Function],
              }
            }
            type={[Function]}
            value={undefined}
          />,
          <Styled(Button)
            bordered={true}
            light={true}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "marginBottom": 12,
              }
            }
            virtual={undefined}
          >
            <Styled(Text)
              virtual={undefined}
            >
               Logg inn 
            </Styled(Text)>
          </Styled(Button)>,
          <Styled(Button)
            bordered={true}
            light={true}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "marginBottom": 12,
              }
            }
            virtual={undefined}
          >
            <Styled(Text)
              virtual={undefined}
            >
              Registrer deg her
            </Styled(Text)>
          </Styled(Button)>,
        ],
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "i18n": Object {
              "add": "Add",
              "down": "↓",
              "optional": " (optional)",
              "remove": "✘",
              "required": "",
              "up": "↑",
            },
            "options": undefined,
            "stylesheet": Object {
              "button": Object {
                "alignSelf": "stretch",
                "backgroundColor": "#48BBEC",
                "borderColor": "#48BBEC",
                "borderRadius": 8,
                "borderWidth": 1,
                "height": 36,
                "justifyContent": "center",
                "marginBottom": 10,
              },
              "buttonText": Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 18,
              },
              "checkbox": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "controlLabel": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
              },
              "dateTouchable": Object {
                "error": Object {},
                "normal": Object {},
              },
              "dateValue": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
              },
              "datepicker": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "errorBlock": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "fieldset": Object {},
              "formGroup": Object {
                "error": Object {
                  "marginBottom": 10,
                },
                "normal": Object {
                  "marginBottom": 10,
                },
              },
              "helpBlock": Object {
                "error": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
                "normal": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
              },
              "pickerContainer": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "open": Object {},
              },
              "pickerTouchable": Object {
                "active": Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#cccccc",
                },
                "error": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
                "normal": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
              },
              "pickerValue": Object {
                "error": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
                "normal": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
              },
              "select": Object {
                "error": Object {},
                "normal": Object {},
              },
              "textbox": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "notEditable": Object {
                  "backgroundColor": "#eeeeee",
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#777777",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
              },
              "textboxView": Object {
                "error": Object {},
                "normal": Object {},
                "notEditable": Object {},
              },
            },
            "templates": Object {
              "checkbox": [Function],
              "datepicker": [Function],
              "list": [Function],
              "select": [Function],
              "struct": [Function],
              "textbox": [Function],
            },
            "type": [Function],
            "value": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": true,
            "children": <Styled(Text)
              virtual={undefined}
            >
               Logg inn 
            </Styled(Text)>,
            "light": true,
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "marginBottom": 12,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Logg inn ",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": " Logg inn ",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": true,
            "children": <Styled(Text)
              virtual={undefined}
            >
              Registrer deg her
            </Styled(Text)>,
            "light": true,
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "marginBottom": 12,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Registrer deg her",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Registrer deg her",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
